diff --git a/libraries/AP_HAL/AP_HAL_Boards.h b/libraries/AP_HAL/AP_HAL_Boards.h
index 449abcddbd4a9..634a526f0e6e3 100644
--- a/libraries/AP_HAL/AP_HAL_Boards.h
+++ b/libraries/AP_HAL/AP_HAL_Boards.h
@@ -52,6 +52,7 @@
 #define HAL_BOARD_SUBTYPE_LINUX_T3_GEM_O1  1029
 #define HAL_BOARD_SUBTYPE_LINUX_RSAXVC_V1  1030
 #define HAL_BOARD_SUBTYPE_LINUX_RPI        1031
+#define HAL_BOARD_SUBTYPE_LINUX_SYSFS_PWM  1032
 
 /* HAL CHIBIOS sub-types, starting at 5000
 
diff --git a/libraries/AP_HAL_Linux/HAL_Linux_Class.cpp b/libraries/AP_HAL_Linux/HAL_Linux_Class.cpp
index 0a20a51134882..af9a40fbc904b 100644
--- a/libraries/AP_HAL_Linux/HAL_Linux_Class.cpp
+++ b/libraries/AP_HAL_Linux/HAL_Linux_Class.cpp
@@ -233,6 +233,10 @@ RCOutput_Multi::RCOutputGroup rcoutSysfsDriver = {2, NEW_NOTHROW RCOutput_Sysfs(
 static RCOutput_Multi rcoutDriver{2, &rcoutPcaDriver, &rcoutSysfsDriver};
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_T3_GEM_O1
 static RCOutput_Sysfs rcoutDriver(0, 0, 7);
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_SYSFS_PWM
+static RCOutput_Sysfs rcoutDriver(HAL_RCOUT_SYSFS_CHIP,
+                                  HAL_RCOUT_SYSFS_CHANNEL_BASE,
+                                  HAL_RCOUT_SYSFS_CHANNEL_COUNT);
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RPI
 static RCOutput_RPI rcoutDriver;
 #else
