diff --git a/soc/silabs/silabs_s2/Kconfig.defconfig b/soc/silabs/silabs_s2/Kconfig.defconfig
index 2e8f986cb80..2a762b33656 100644
--- a/soc/silabs/silabs_s2/Kconfig.defconfig
+++ b/soc/silabs/silabs_s2/Kconfig.defconfig
@@ -100,7 +100,7 @@ configdefault OPENTHREAD_MIN_RECEIVE_ON_AFTER
 	default 0
 
 configdefault OPENTHREAD_MAC_SOFTWARE_TX_SECURITY_ENABLE
-	default n
+	default y
 
 configdefault OPENTHREAD_IP6_MAX_EXT_MCAST_ADDRS
 	default 4
diff --git a/drivers/ieee802154/ieee802154_silabs_efr32.c b/drivers/ieee802154/ieee802154_silabs_efr32.c
index 59d16289504..bb53a9385f6 100644
--- a/drivers/ieee802154/ieee802154_silabs_efr32.c
+++ b/drivers/ieee802154/ieee802154_silabs_efr32.c
@@ -1961,8 +1961,7 @@ static enum ieee802154_hw_caps silabs_efr32_get_capabilities(const struct device
 	 * rx slot handling. Add them when CSL support lands.
 	 */
 	return IEEE802154_HW_FCS | IEEE802154_HW_FILTER | IEEE802154_HW_TX_RX_ACK |
-	       IEEE802154_HW_ENERGY_SCAN | IEEE802154_HW_SLEEP_TO_TX | IEEE802154_HW_CSMA |
-	       IEEE802154_HW_TX_SEC;
+	       IEEE802154_HW_ENERGY_SCAN | IEEE802154_HW_SLEEP_TO_TX | IEEE802154_HW_CSMA;
 }
 
 static int silabs_efr32_cca(const struct device *dev)
